pkg/kn/flags/channel_types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flags
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"knative.dev/client/pkg/kn/config"
+)
+
+type ChannelTypeFlags struct {
+	ctype string
+}
+
+func (i *ChannelTypeFlags) Add(f *pflag.FlagSet) {
+	f.StringVar(&i.ctype,
+		"type",
+		"",
+		"Override channel type to create, in the format '--type Group:Version:Kind'. "+
+			"If flag is not specified, it uses default messaging layer settings for channel type, cluster wide or specific namespace. "+
+			"You can configure aliases for channel types in kn config and refer the aliases with this flag. "+
+			"You can also refer inbuilt channel type InMemoryChannel using an alias 'imc' like '--type imc'. "+
+			"Examples: '--type messaging.knative.dev:v1alpha1:KafkaChannel' for specifying explicit Group:Version:Kind.")
+
+	for _, p := range config.GlobalConfig.ChannelTypeMappings() {
+		//user configration might override the default configuration
+		ctypeMappings[p.Alias] = schema.GroupVersionKind{
+			Kind:    p.Kind,
+			Group:   p.Group,
+			Version: p.Version,
+		}
+	}
+}
+
+// ctypeMappings maps aliases used for channel types to their GroupVersionKind
+var ctypeMappings = map[string]schema.GroupVersionKind{
+	"imc": {
+		Group:   "messaging.knative.dev",
+		Version: "v1beta1",
+		Kind:    "InMemoryChannel",
+	},
+}
```

Comment:
We'll ship channel management with kn in 0.17. We're working with v1beta1 APIs/clientset of eventing in client, to be compatible with 0.17 and earlier releases.
We create channels and defer the type to cluster/namespace defaults unless specified explicitly on cli using `--type` flag.
For convenience, we're giving this short hand alias for IMC channels, to override if user wants to, and pinning its GVK at `v1beta1`.
Looking at https://github.com/knative/eventing/pull/3793 and if the cluster default is going to be IMC at v1 for 0.17, should we rename this alias and add another at v1 ?
like `imcv1` and `imcv1beta1`, this way user get to choose whichever is available in the cluster

wdyt @matzew @rhuss

Reply:
adding explicit aliases: imcv1 and imcv1beta1

Reply:
what's about having a `--type imc` alias for v1 and `--type imc:v1beta1` for v1beta1 (and for completeness sake also a `--type imc:v1` ?)

Reply:
reasoning behind this is, that I think we should separate the version here more visually and also make it optional so that we have a short default.  I expect that v1beta1 is only temporary and will fade away over time ...